embedchain/embedder/openai.py

```diff
@@ -16,16 +16,18 @@ def __init__(self, config: Optional[BaseEmbedderConfig] = None):
         if self.config.model is None:
             self.config.model = "text-embedding-ada-002"
 
+        api_key = self.config.api_key or os.environ["OPENAI_API_KEY"]
+
         if self.config.deployment_name:
             embeddings = AzureOpenAIEmbeddings(deployment=self.config.deployment_name)
             embedding_fn = BaseEmbedder._langchain_default_concept(embeddings)
         else:
-            if os.getenv("OPENAI_API_KEY") is None and os.getenv("OPENAI_ORGANIZATION") is None:
+            if api_key is None and os.getenv("OPENAI_ORGANIZATION") is None:
```

Comment:
This should check for `api_key`?

Reply:
Yes which is either `config.api_key or env.api_key`